woocommerce/woocommerce-ios
# Update SiteVisitStats to use Storage fmwk + ResultsController

This PR partially addresses #258 by doing the following:

* **Networking:** Updated the`SiteVisitStats` & `SiteVisitStatsItem` models — `MIContainer` is no longer stored as a var, instead we now decode the individual fields immediately when the incoming JSON is parsed
* **Storage:** Created data model v4 which now contains the `SiteVisitStats` and `SiteVisitStatsItem` entities
* **Storage:** Added `loadSiteVisitStats()` to `StorageType+Extensions`
* **Yosemite:** Added `StorageSiteVisitStats` and `StorageSiteVisitStatsItem` models
* **Yosemite:** Updated `StatsStore` to now upsert `SiteVisitStats` when retrieved from the server
* **Yosemite:** Updated `StatsAction.retrieveSiteVisitStats()` action to no longer return `SiteVisitStats`
* **WooCommerce:** `PeriodDataViewController` now uses a `ResultsController` for the Visitors data
* **WooCommerce:** Simplified `StoreStatsViewController`'s `syncVisitorStats` function
* Updated a bunch of unit tests around `SiteVisitStats`

## Testing

0. Make sure everything builds without issue
1. Verify the unit tests are ✅ 
2. Run the app - make sure everything looks ok the visitors field
3. Close the app, visit the store you are logged into, and open the app again (maybe pull to refresh) - verify the visits count updates

@jleandroperez Could you take a 👀 at this PR? Thanks!!! 🎸 🥇
